library/std/src/sys/pal/windows/net.rs

```diff
@@ -572,3 +574,22 @@ impl FromRawSocket for Socket {
         unsafe { Self(FromRawSocket::from_raw_socket(raw_socket)) }
     }
 }
+
+pub fn gethostname() -> io::Result<OsString> {
+    init();
+
+    // The documentation of GetHostNameW says that a buffer size of 256 is
+    // always enough.
+    let mut buffer = [0; 256];
+    // SAFETY: these parameters specify a valid, writable region of memory.
+    let ret = unsafe { c::GetHostNameW(buffer.as_mut_ptr(), buffer.len() as i32) };
```

Comment:
I guess my one nitpick is that maybe the max buffer len could be a constant instead of using `as i32`, which is often a code smell but in this case it's narrowly scoped enough that I don't feel strongly about it.

Reply:
It would be easiest to keep the `as i32` here, since if someone goes to implement this on Windows 7, they'd probably need it anyways.